daemon/graphdriver/overlay2/overlay.go

```diff
@@ -312,23 +322,33 @@ func (d *Driver) Cleanup() error {
 // CreateReadWrite creates a layer that is writable for use as a container
 // file system.
 func (d *Driver) CreateReadWrite(id, parent string, opts *graphdriver.CreateOpts) error {
+	if opts != nil && len(opts.StorageOpt) != 0 && !projectQuotaSupported {
+		return fmt.Errorf("--storage-opt is supported only for overlay over xfs with 'pquota' mount option")
+	}
+
+	if opts == nil {
+		opts = &graphdriver.CreateOpts{}
+	}
+
+	if len(opts.StorageOpt) == 0 {
+		opts.StorageOpt = map[string]string{
+			"size": fmt.Sprintf("%d", d.options.quota.Size),
```

Comment:
This assumption of "size" param was in the old code. The parsing of the options is done in Create. So In CreateReadWrite we do not know what the "options" are (we would need to parse and unparse) and in Create we do not know if it is for readWrite only. 

Create is called for read layers and readwrite layers. The current discussions evolved to having "the size" option only for readwrite layer. Hence the changes.